### PR TITLE
PORT-10837 ArgoCD Integration guide refactoring

### PR DESCRIPTION
**Pull request by kodjomiles**
# Description

Restructure the documentation structure and made some refactoring

## Added docs pages
- Examples page (`/build-your-software-catalog/sync-data-to-catalog/argocd/example`)

## Updated docs pages

Please also include the path for the updated docs

- ArgoCD (`/build-your-software-catalog/sync-data-to-catalog/argocd`)

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 1]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
-Port's ArgoCD integration allows you to import `cluster`, `project`, `application`, `deployment-history`, `kubernetes-resource` and `managed-resource` from your ArgoCD instance into Port, according to your mapping and definition.
+Port's ArgoCD integration allows you to model ArgoCD resources in your software catalog and ingest data into them.
+
+
+## Overview
+
+This integration allows you to:
+
+- Map and organize your desired ArgoCD resources and their metadata in Port (see supported resources below).
+- Watch for X object changes (create/update/delete) in real-time, and automatically apply the changes to your software catalog.
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:25]:
```suggestion
- Watch for ArgoCD object changes (create/update/delete) in real-time, and automatically apply the changes to your software catalog.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 2]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
+
+- [`cluster`](https://cd.apps.argoproj.io/swagger-ui#operation/ClusterService_List)
+- [`project`](https://cd.apps.argoproj.io/swagger-ui#operation/ProjectService_List)
+- [`application`](https://cd.apps.argoproj.io/swagger-ui#operation/ApplicationService_List)
+- [`deployment-history`](https://cd.apps.argoproj.io/swagger-ui#operation/ApplicationService_List) 
+- [`kubernetes-resource`](https://cd.apps.argoproj.io/swagger-ui#operation/ApplicationService_List)
+- [`managed-resource`](https://cd.apps.argoproj.io/swagger-ui#operation/ApplicationService_ManagedResources)
 
-## Installation
+## Prerequisite
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:39]:
```suggestion
## Prerequisites
```

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 3]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
+
+### Generating ArgoCD token
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:41]:
```suggestion
### Generate an ArgoCD token
```

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 4]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
+
+1. Navigate to `<serverURL>/settings/accounts/<user>`. For example, if you access your ArgoCD at `https://localhost:8080`, you should navigate to `https://localhost:8080/settings/accounts/<user>`
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:43]: this section is crowded, put an empty line between each item in the list

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 5]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
+2. The user should have `apiKey` capabilities to allow generating authentication tokens for API access. If you don't have a user created yet, follow the guide on [how to create a new ArgoCD user](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/#create-new-user)
+3. Newly created users may have limited scope to resources by default. For that reason, You will need to configure RBAC policy for the new user by following [this guide](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/)
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:45]:
```suggestion
3. Newly created users may have limited scope to resources by default. For that reason, You will need to configure the RBAC policy for the new user by following [this guide](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/)
```

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 6]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
+4. Ensure that the policy definition grants enough permission to `read` resources such as `applications`, `clusters`, `projects`, `repositories` etc.
+5. Under **Tokens** on your ArgoCD UI, Click **Generate New** to create a new token for the user or use the CLI:
+
+```bash
+argocd account generate-token --account <username>
+```
+
+:::tip Creating ArgoCD user with readonly permissions
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:53]: this section is not clear.
1. As a reader, I don't know by reading this if I need to create a readonly user or not, there is no context.
2. Why is this in an admonition?

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 7]
```diff
@@ -8,66 +9,130 @@ import ArgoCDWebhookConfig from '/docs/build-your-software-catalog/custom-integr
...
 
 Choose one of the following installation methods:
 
 <Tabs groupId="installation-methods" queryString="installation-methods">
 
-<TabItem value="real-time-always-on" label="Real Time & Always On" default>
+<TabItem value="real-time-self-hosted" label="Real-time (self-hosted)" default>
 
-Using this installation option means that the integration will be able to update Port in real time using webhooks.
+<h2> Prerequisites </h2>
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:124]: missing this line above the prerequisites:
"Using this installation method means that the integration will be able to update Port in real time using webhooks."

You can see that we have it here:
https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/other/backstage

Please add this line to the template if it is not already there.

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 8]
```diff
@@ -358,210 +442,19 @@ ingest_data:
...
-### Prerequisite
-
-1. You have access to a Kubernetes cluster where ArgoCD is deployed.
-2. You have `kubectl` installed and configured to access your cluster.
-
-### Steps
-
 1. Install ArgoCD notifications manifest;
 
 ```bash showLineNumbers
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:460]: for code snippets that have only one line, do not add "showLineNumbers"

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/example.md [hunk 9]
```diff
@@ -0,0 +1,562 @@
+---
+sidebar_position: 2
+---
+
+# Examples
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/example.md:5]: please change the name of this file to "examples.md" (and not example.md). 
Make sure to fix the link in the "argocd.md" page accordingly

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md [hunk 10]
```diff
@@ -358,210 +442,19 @@ ingest_data:
...
-        entity:
-          mappings: ...
-  ```
-
-  :::tip Blueprint key
-  Note the value of the `blueprint` key - if you want to use a hardcoded string, you need to encapsulate it in 2 sets of quotes, for example use a pair of single-quotes (`'`) and then another pair of double-quotes (`"`)
-  :::
-
-## Configuring real-time updates
+### Configuring real-time updates
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:454]:
```suggestion
### Configure real-time updates
```

**Review by hadar-co — CHANGES_REQUESTED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md (hunk 6 above)
Comment (kodjomiles) [reply docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:53]: I just did the rearrangement. So I took the entire content and dumped it in there and reading as a user I understood it as if that was the main step

File: docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md (hunk 7 above)
Comment (kodjomiles) [reply docs/build-your-software-catalog/sync-data-to-catalog/argocd/argocd.md:124]: it was  not in the template. and I didn't add it to the pagerduty too. is something we missed or its intentional

**Review by hadar-co — APPROVED** [review]

**Pull request merged**